pkg/translate/translateValue.go

```diff
@@ -0,0 +1,554 @@
+				return err
+			}
+			continue
+		}
+
+		if k8sSettingName == "env" {
+			err := translateEnv(v.outPath, m, cpSpecTree)
+			if err != nil {
+				return fmt.Errorf("error in translating k8s Env: %s", err.Error())
+			}
+			if _, err := DeleteFromTree(valueTree, util.ToYAMLPath(inPath), util.ToYAMLPath(inPath)); err != nil {
+				return err
+			}
+			continue
+		}
+
+		if k8sSettingName == "podAntiAffinityLabelSelector" || k8sSettingName == "podAntiAffinityTermLabelSelector" {
+			err := translateAffinity(v.outPath, m, cpSpecTree)
+			if err != nil {
+				return fmt.Errorf("error in translating k8s Affinity: %s", err.Error())
+			}
+			if _, err := DeleteFromTree(valueTree, util.ToYAMLPath(inPath), util.ToYAMLPath(inPath)); err != nil {
+				return err
+			}
+			continue
+		}
+
+		if mstr, ok := m.(string); ok && mstr == "" {
+			log.Infof("path %s is empty string, skip mapping.", inPath)
+			continue
+		}
+		// Zero int values are due to proto3 compiling to scalars rather than ptrs. Skip these because values of 0 are
+		// the default in destination fields and need not be set explicitly.
+		if mint, ok := util.ToIntValue(m); ok && mint == 0 {
+			log.Infof("path %s is int 0, skip mapping.", inPath)
+			continue
+		}
+
+		output := util.ToYAMLPath(v.outPath)
+		log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", output)
+
+		if err := tpath.WriteNode(cpSpecTree, output, m); err != nil {
+			return err
+		}
+
+		if _, err := DeleteFromTree(valueTree, util.ToYAMLPath(inPath), util.ToYAMLPath(inPath)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// translateRemainPaths translates remaining paths that are not availing in existing mappings.
+func (t *ValueYAMLTranslator) translateRemainPaths(valueTree map[string]interface{},
+	cpSpecTree map[string]interface{}, path util.Path) error {
+	for key, val := range valueTree {
+		newPath := append(path, key)
+		// value set to nil means no translation needed or being translated already.
+		if val == nil {
+			continue
+		} else {
+			switch node := val.(type) {
+			case map[string]interface{}:
+				err := t.translateRemainPaths(node, cpSpecTree, newPath)
+				if err != nil {
+					return err
+				}
+			case []interface{}:
+				errs := util.Errors{}
+				for _, newNode := range node {
+					newMap, ok := newNode.(map[string]interface{})
+					if !ok {
+						return fmt.Errorf("fail to convert node to map[string] interface")
+					}
+					errs = util.AppendErr(errs, t.translateRemainPaths(newMap, cpSpecTree, newPath))
+				}
+				if errs != nil {
+					return errs
+				}
+			// remaining leaf need to be put into common.values
+			default:
+				err := t.translateToCommonValues(cpSpecTree, newPath, val)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// DeleteFromTree sets value at path of input tree to nil
+func DeleteFromTree(valueTree map[string]interface{}, path util.Path, remainPath util.Path) (bool, error) {
```

Comment:
I think you could also use tpath here. See tpath_test.go for delete examples. Maybe you can create a wrapper called DeleteNode in that package.

Reply:
changed to use writeNode directly

Reply:
Actually the writeNode still doesn't work here, it looked "successful" because of the test setup issue in that revision